lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+--
+--
+-- This module allows the wallet to retrieve blocks from a known @Jormungandr@
+-- node. This is done by providing a @NetworkLayer@ with some logic building on
+-- top of an underlying @Jormungandr@ HTTP client.
+module Cardano.Wallet.Jormungandr.Network
+    ( Jormungandr (..)
+    , mkJormungandr
+
+    -- * Re-export
+    , BaseUrl (..)
+    , newManager
+    , defaultManagerSettings
+    , Scheme (..)
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Jormungandr.Api
+    ( BlockId, GetTipId, api )
+import Cardano.Wallet.Network
+    ( ErrNetworkUnreachable (..) )
+import Control.Exception
+    ( Exception )
+import Control.Monad.Catch
+    ( throwM )
+import Control.Monad.Trans.Except
+    ( ExceptT (..) )
+import Data.Proxy
+    ( Proxy (..) )
+import Network.HTTP.Client
+    ( Manager, defaultManagerSettings, newManager )
+import Servant.Client
+    ( BaseUrl (..), ClientM, Scheme (..), client, mkClientEnv, runClientM )
+import Servant.Client.Core
+    ( ServantError (..) )
+import Servant.Links
+    ( Link, safeLink )
+
+-- TODO: Implement a NetworkLayer
+-- -- | Constructs a network layer with the given @Jormungandr@ client.
+-- mkNetworkLayer :: Monad m => Jormungandr m -> NetworkLayer m
+-- mkNetworkLayer _httpBridge = NetworkLayer
+--     { nextBlocks = undefined
+--     , networkTip = undefined
+--     , postTx = undefined
+--     }
+--
+-- -- | Creates a jormungandr 'NetworkLayer' using the given connection
+-- -- settings.
+-- newNetworkLayer
+--     :: Int -> IO (NetworkLayer IO)
+-- newNetworkLayer port = mkNetworkLayer <$> newHttpBridge port
+--   where
+--     newHttpBridge = undefined
+
+{-------------------------------------------------------------------------------
+                            Jormungandr Client
+-------------------------------------------------------------------------------}
+
+-- | Endpoints of the jormungandr REST API.
+newtype Jormungandr m = Jormungandr
+    { getTipId
+        :: ExceptT ErrNetworkUnreachable m BlockId
```

Comment:
When we implement [`networkTip`](https://github.com/input-output-hk/cardano-wallet/blob/80bf249a03b40811c818094df8ad42b00041fd89/lib/core/src/Cardano/Wallet/Network.hs#L39) of `NetworkLayer`, we will have to use `ErrNetworkTip` instead of `ErrNetworkUnreachable` to conform. But as the endpoint is only expected to return 200, I see no reason to use `ErrNetworkTip` now.

Reply:
So it might be worth looking into removing this error from the `http-bridge` too then and maybe just, work-around by re-trying and throwing after `x` retries. It's not normal for the network to not be available anyway :thinking: ...

Reply:
There is essentially already retry logic on the call-sites
- https://github.com/input-output-hk/cardano-wallet/blob/f2abd85c5bdc0d69ed31a32a2e2020c923290343/lib/core/src/Cardano/Wallet.hs#L477
- https://github.com/input-output-hk/cardano-wallet/blob/f2abd85c5bdc0d69ed31a32a2e2020c923290343/lib/core/src/Cardano/Wallet.hs#L436

It seems to me that we can't "just move it" to the `NetworkLayer`. And I cannot picture why having additional retrying would be helpful, except to make our integration tests more accepting on ci.

Experiment -> f2abd85c5bdc0d69ed31a32a2e2020c923290343